waspc/src/StrongPath.hs

```diff
@@ -1,60 +1,104 @@
 module StrongPath
     ( Path
-    , Abs, File, Dir
+    , Abs, Rel, Dir, File
+    , parseRelDir, parseRelFile, parseAbsDir, parseAbsFile
+    , toFilePath
     , (</>)
-    , fromRelDir, fromRelFile, fromAbsDir, fromAbsFile
-    , toRelDir, toRelFile, toAbsDir, toAbsFile
+    , castRel
+    , parent
+    , fromPathRelDir, fromPathRelFile, fromPathAbsDir, fromPathAbsFile
+    , toPathRelDir, toPathRelFile, toPathAbsDir, toPathAbsFile
     ) where
 
 import qualified Path as P
+import Control.Monad.Catch (MonadThrow)
 
 
-data Path from to = RelDir  (P.Path P.Rel P.Dir)
-                  | RelFile (P.Path P.Rel P.File)
-                  | AbsDir  (P.Path P.Abs P.Dir)
-                  | AbsFile (P.Path P.Abs P.File)
+data Path b t = RelDir  (P.Path P.Rel P.Dir)
```

Comment:
what does b and t mean? I like it better than from/to, that wasn't totally applicable

Reply:
I took b and t from Path library. They say b is for base and t is for type.

Reply:
cool, makes sense